GCM: don't validate for aligned plain/ciphertext

jdlcdl opened the pull request:
GCM doesn't require aligned-blocks, so check is avoided.

Prior to the following "krux_encryption" commit, These GCM changes were working fine only because padding to aligned aes-blocks was being done for GCM as it is for ECB/CBC.  However, since the following change, for plaintext/ciphertext that is not aligned, GCM would have been broken.

https://github.com/selfcustody/krux/pull/546/commits/95fc5580847f8552b51e12bc6a13e3bacd7197c7

Since making the changes in this MaixPy pr, I'm no longer having problems with GCM (which is no longer padded in "krux_encryption" branch (pr:456)) and not having problems between device and  simulator using pycryptodome.

jdlcdl committed:
GCM: don't validate for aligned plain/ciphertext

tadeubas commented:
Wow, nice finding! I've compared this Maixpy code with [Loboris k210 cryptolib](https://github.com/loboris/MicroPython_K210_LoBo/blob/master/k210-freertos/mpy_support/moducryptolib_k210.c) and found that few changes were made on our Maixpy version, addittion of `gcm_tag` and functions `ucryptolib_aes_get_tag` and `ucryptolib_aes_verify_tag`.

I think we could just remove the `const mp_obj_module_t mp_module_ucryptolib;` that is defined 2 times in the file, [at top (line 64 without any value)](https://github.com/selfcustody/MaixPy/blob/7104fb6ff6e1ddda8debd05de42d5557afac0055/components/micropython/port/src/moducryptolib_maix.c#L64)  and at [bottom (line 281 with values)](https://github.com/selfcustody/MaixPy/blob/7104fb6ff6e1ddda8debd05de42d5557afac0055/components/micropython/port/src/moducryptolib_maix.c#L281).

Could you do this change @jdlcdl ?

jdlcdl committed:
removed redundant declaration of mp_module_ucryptolib
  it was declared as a constant at the top of the file and
  also declared and defined below; removed top declaration.

odudex commented:
Great work guys!
